sdk/samples/core/rpc_client_pending_commands.h

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Definition of #az_mqtt5_rpc_client_sample pending command functions. You use the RPC
+ * client to send commands.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _RPC_CLIENT_PENDING_COMMANDS_H
+#define _RPC_CLIENT_PENDING_COMMANDS_H
+
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+// Application can define this value to control how much memory is used to track pending commands
+// (and how many can be in flight at once)
+#ifndef RPC_CLIENT_MAX_PENDING_COMMANDS
+#define RPC_CLIENT_MAX_PENDING_COMMANDS 5
+#endif
+
+typedef struct pending_command
+{
+  az_span correlation_id;
+  az_context context;
+  az_span command_name;
+} pending_command;
+
+typedef struct pending_commands_array
+{
+  int32_t pending_commands_count;
+  pending_command commands[RPC_CLIENT_MAX_PENDING_COMMANDS];
+} pending_commands_array;
+
+AZ_INLINE az_result pending_commands_array_init(
+    pending_commands_array* pending_commands,
+    uint8_t correlation_id_buffers[RPC_CLIENT_MAX_PENDING_COMMANDS]
+                                  [AZ_MQTT5_RPC_CORRELATION_ID_LENGTH])
+{
+  pending_commands->pending_commands_count = 0;
+  for (int i = 0; i < RPC_CLIENT_MAX_PENDING_COMMANDS; i++)
+  {
+    pending_commands->commands[i].correlation_id
+        = az_span_create(correlation_id_buffers[i], AZ_MQTT5_RPC_CORRELATION_ID_LENGTH);
+    az_span_fill(pending_commands->commands[i].correlation_id, 0x0);
+    pending_commands->commands[i].command_name = AZ_SPAN_EMPTY;
+  }
+  return AZ_OK;
+}
+
+AZ_INLINE az_result clock_msec(int64_t* out_clock_msec)
+{
+#ifdef _WIN32
+
+  _az_PRECONDITION_NOT_NULL(out_clock_msec);
+  *out_clock_msec = GetTickCount64();
+#else
+  _az_PRECONDITION_NOT_NULL(out_clock_msec);
+  struct timespec curr_time;
+
+  if (clock_getres(CLOCK_BOOTTIME, &curr_time) == 0) // Check if high-res timer is available
+  {
+    clock_gettime(CLOCK_BOOTTIME, &curr_time);
+    *out_clock_msec = ((int64_t)curr_time.tv_sec * 1000) + ((int64_t)curr_time.tv_nsec / 1000000);
+  }
+  else
+  {
+    // NOLINTNEXTLINE(bugprone-misplaced-widening-cast)
```

Comment:
Remove?

Reply:
Or add a comment above explaining meaning.

Reply:
This is from the posix platform implementation of the clock
https://github.com/Azure/azure-sdk-for-c/blob/feature/v2/sdk/src/azure/platform/az_posix.c#L52